identify/identify.py

```diff
@@ -61,15 +62,27 @@ def tags_from_path(path: str) -> Set[str]:
         tags.add(NON_EXECUTABLE)
 
     # As an optimization, if we're able to read tags from the filename, then we
-    # don't peek at the file contents.
+    # don't peek at the file contents unless the file extension requires it
     t = tags_from_filename(os.path.basename(path))
     if len(t) > 0:
         tags.update(t)
-    else:
-        if executable:
-            shebang = parse_shebang_from_file(path)
-            if len(shebang) > 0:
-                tags.update(tags_from_interpreter(shebang[0]))
+
+    ext = os.path.splitext(os.path.split(path)[-1])[-1].lstrip('.')
+    if (
+        not len(t) and executable
+    ) or ext in extensions.EXTENSIONS_NEED_INTERPRETER_CHECK:
+        try:
+            tags.update(extensions.EXTENSIONS_NEED_INTERPRETER_CHECK[ext])
+        except KeyError:
+            pass
+
+        shebang = parse_shebang_from_file(path)
+        if len(shebang) > 0:
+            tags.update(
+                tags_from_interpreter(
+                    shebang[0].split('|')[0].strip(),
+                ),
+            )
```

Comment:
The splitting of the `|` delimited interpreters may be better located in the `_shebang_split` function.. I can move there if this looks good.

Reply:
If the `|` convention to separate multiple interpreters is non-standard and only used with one particular file type, I don't know about trying to parse it on all files. I can't find much Googling about other uses, but the [one](http://reasonableapproximation.net/2010/07/27/silly-things-to-do-with-shebang-lines.html) usage I can find has different syntax than salt uses. But that's ultimately up to @asottile .